system-int-test/src/cmr/system_int_test/utils/service_util.clj

```diff
@@ -0,0 +1,120 @@
+   [cmr.common.mime-types :as mt]
+   [cmr.common.util :as util]
+   [cmr.mock-echo.client.echo-util :as echo-util]
+   [cmr.system-int-test.data2.core :as d]
+   [cmr.system-int-test.system :as s]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest-util]
+   [cmr.system-int-test.utils.metadata-db-util :as mdb]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.transmit.echo.tokens :as tokens]
+   [cmr.transmit.service :as transmit-service]))
+
+(defn assert-convert-kebab-case
+  [data]
+  ; XXX This can be uncommented once service support is added to the metadata db
+  ;     See CMR-4172
+  ; (util/assert-convert-kebab-case [:concept-id :revision-id
+  ;                                  :service-name :originator-id]
+  ;                                 data))
+  true)
+
+(defn grant-all-service-fixture
+  "A test fixture that grants all users the ability to create and modify
+  services."
+  [f]
+  (echo-util/grant-all-service (s/context))
+  (f))
+
+(def sample-service
+  {:Name "A name"
+   :Description "A UMM-S description"
+   :Format ""
+   :Type "Web Mapping Service"
+   :Version "1.1.1"})
+
+(defn make-service
+  "Makes a valid service based on the given input."
+  ([]
+   (make-service nil))
+  ([attrs]
+   (merge sample-service attrs))
+  ([index attrs]
+   (merge
+    sample-service
+    {:Name (str "Name" index)
+     :Version (str "V" index)
+     :Description (str "UMM-S description " index)}
+    attrs)))
+
+(defn create-service
+  "Creates a service."
+  ([token service]
+   (create-service token service nil))
+  ([token service options]
+   (let [options (merge {:raw? true :token token} options)]
+     (ingest-util/parse-map-response
+      (transmit-service/create-service (s/context) service options)))))
+
+(defn update-service
+  "Updates a service."
+  ([token service]
+   (update-service token (:service-id service) service nil))
+  ([token service-id service]
+   (update-service token service-id service nil))
+  ([token service-id service options]
+   (let [options (merge {:raw? true :token token} options)]
+     (ingest-util/parse-map-response
+      (transmit-service/update-service (s/context) service-id service options)))))
+
+(defn save-service
+  "A helper function for creating or updating services for search tests.
+
+   If the service does not have a :concept-id, it saves it. If the service has
+   a :concept-id, it updates the service. Returns the saved service along
+   with :concept-id, :revision-id, :errors, and :status."
+  [token service]
+  (let [service-to-save (select-keys service [:service-name :description :revision-date])
+        response (if-let [concept-id (:concept-id service)]
+                   (update-service token (:service-name service) service-to-save)
+                   (create-service token service-to-save))
+        service (-> service
+                    (update :service-name string/lower-case)
```

Comment:
No clue; this was copied from the work you did in `cmr.system-int-test.utils.variable-util/save-variable`.